extension/src/popup/views/AccountCreator.tsx

```diff
@@ -76,70 +94,95 @@ export const AccountCreator = () => {
+                        }
+                        {...field}
+                      />
+                    )}
+                  </Field>
                 </FormRow>
                 <FormRow>
-                  <TextField
-                    autoComplete="off"
-                    name="confirmPassword"
-                    placeholder="Confirm password"
-                    type="password"
-                  />
-                  <Error name="confirmPassword" />
+                  <Field name="confirmPassword">
+                    {({ field }: FieldProps) => (
+                      <Input
+                        autoComplete="off"
+                        id="confirm-password-input"
+                        placeholder="Confirm password"
+                        type="password"
+                        error={
+                          authError ||
+                          (errors.confirmPassword && touched.confirmPassword)
+                            ? errors.confirmPassword
+                            : null
+                        }
+                        {...field}
+                      />
+                    )}
+                  </Field>
                 </FormRow>
+
                 <PasswordRequirements />
                 <FormRow>
-                  <CheckboxField
-                    name="termsOfUse"
-                    label={
-                      <span>
-                        I have read and agree to{" "}
-                        <a href="https://stellar.org/terms-of-service">
-                          Terms of Use
-                        </a>{" "}
-                        and{" "}
-                        <a href="https://stellar.org/privacy-policy">
-                          Privacy Policy
-                        </a>
-                      </span>
-                    }
-                  />
-                  <Error name="termsOfUse" />
+                  <Field name="termsOfUse">
+                    {({ field }: FieldProps) => (
+                      <Checkbox
+                        autoComplete="off"
+                        id="termsOfUse-input"
+                        label={
+                          <span>
+                            I have read and agree to{" "}
+                            <TextLink
+                              variant={TextLink.variant.secondary}
+                              href="https://stellar.org/terms-of-service"
+                            >
+                              Terms of Use
+                            </TextLink>
+                          </span>
+                        }
+                        {...field}
+                      />
+                    )}
+                  </Field>
                 </FormRow>
-                <FormRow>
+                {errors.termsOfUse && touched.termsOfUse
+                  ? errors.termsOfUse
+                  : null}
+                <ButtonRowEl>
                   <SubmitButton
```

Comment:
Should we use SDS `Button` component here?

Reply:
good catch I wasn't sure about that. @piyalbasu or thoughts on changing `SubmitButton` to use the SDS button? since they all should be using SDS from my understanding. I can change in this PR if you agree

Reply:
@acharb Yeah you should use the SDS Button. In fact, by the end of the reskin, we'll be removing `SubmitButton` completely from the repo as we'll have replaced all instances with the SDS Button.

Instead of making any changes to `SubmitButton`, I would just import `Button` from SDS in this component, similar to what I did here: https://github.com/stellar/freighter/blob/release/2.0.0/extension/src/popup/views/SignTransaction.tsx

Reply:
https://github.com/stellar/freighter/pull/357/commits/8be89e5cda99382b15082fbaefdd954ad22e5006